File: tvn24.py
import database
import requests
from bs4 import BeautifulSoup



def scraping_tvn24_pages():
    res = requests.get("https://tvn24.pl/tvn24-news-in-english")
    soup = BeautifulSoup(res.text, "html.parser")

    title = soup.find_all(class_='teaser-wrapper__content default-teaser default-teaser--standard')

    time = soup.find_all(class_='label-date')

    newslist = filtering_tvn24_news(title , time)
    title = []
    titlelink = []
    time = []
    detaillink = []
    detail = []
    for news in newslist:
        title.append(news["title"].replace("'", '"'))
        titlelink.append(news['title_link'].replace("'", '"'))
        time.append(news['time'].replace("'", '"'))
        detaillink.append(news['detaillink'].replace("'", '"'))
        detail.append(news['detail'].replace("'", '"'))
    database.insert_into_database("tvn24", title, titlelink, time, detaillink, detail)
    return "tvn24 database updated"


def filtering_tvn24_news(title, times):
    news_list = []
    for index, item in enumerate(title[:8]):
        title = item.find('h2').get_text()
        title_link = item.find('a').get('href')
        detail = item.find('p').get_text()
        time = times[index].get_text()

        match_found = 0
        for tit in news_list:
            header = tit["title"]
            if header == title:
                match_found = 1
                continue
        if match_found == 0:
            current_news = {'title': title, 'title_link': title_link, 'time' : time,'detaillink':title_link,'detail': detail}
            news_list.append(current_news)

    return news_list[:5]

File: frontend/src/components/tvn24/Tvn24.js
import React from "react";
import { FcClock } from "react-icons/fc";

import './Tvn24.css';

const Tvn24 = ({ newsList }) => {
    // console.log(newsList);
    return(
        <div className="">
            <h2>Tvn24</h2>
           <div>
               <a className="title" href={newsList.titlelink[0]}>{newsList.title[0]}</a>
            </div>
            <div>
                <a className="detail" href={newsList.detaillink[0]}>{newsList.detail[0]}</a><br/>
                <h3><FcClock className="mr2" />{newsList.time[0]}</h3>
            </div>
            <div>
               <a className="title" href={newsList.titlelink[1]}>{newsList.title[1]}</a>
            </div>
            <div>
                <a className="detail" href={newsList.detaillink[1]}>{newsList.detail[1]}</a><br/>
                <h3><FcClock className="mr2" /> {newsList.time[1]}</h3>
            </div><div>
               <a className="title" href={newsList.titlelink[2]}>{newsList.title[2]}</a>
            </div>
            <div>
                <a className="detail" href={newsList.detaillink[2]}>{newsList.detail[2]}</a><br/>
                <h3><FcClock className="mr2" />{newsList.time[2]}</h3>
            </div><div>
               <a className="title" href={newsList.titlelink[3]}>{newsList.title[3]}</a>
            </div>
            <div>
                <a className="detail" href={newsList.detaillink[3]}>{newsList.detail[3]}</a><br/>
                <h3><FcClock className="mr2" />{newsList.time[3]}</h3>
            </div><div>
               <a className="title" href={newsList.titlelink[4]}>{newsList.title[4]}</a>
            </div>
            <div>
                <a className="detail" href={newsList.detaillink[4]}>{newsList.detail[4]}</a><br/>
                <h3><FcClock className="mr2" /> {newsList.time[4].slice(-4)}</h3>
            </div>
        </div>
    )
}

export default Tvn24;